ref/general/threading.adoc

```diff
@@ -0,0 +1,92 @@
+
+. Uses historical data to predict whether performance would be better at smaller or larger pool sizes. If there is no historical data for the smaller or larger pool size, predicts performance by random choice.
+
+. Adjusts pool size (increase, decrease, or unchanged) based on predicted performance, within bounds specified in settings.
+
+. Goes back to sleep.
+
+There are a various factors other than the thread pool size that can affect throughput in the Open Liberty server.
+The relationship between pool size and observed throughput is not perfectly smooth or continuous.
+Therefore, to improve the predictions derived from the historical throughput data, the controller considers not just the closest larger and smaller pool size performance, but also includes several increments in each direction.
+
+=== Resolving hangs
+
+In some application scenarios, all the threads in the pool can become blocked by tasks that must wait for other work to finish before they can run.
+In these cases, the server can become hung at a certain pool size.
+To resolve this type of situation, the thread pool controller enters a hang resolution mode.
+
+Hang resolution adds threads to the pool to allow the server to resume normal execution.
+Hang resolution also shortens the controller cycle duration to break out of the deadlock quickly.
+
+When the controller observes that tasks are being completed again, normal operation resumes.
+The controller cycle returns to its normal duration, and pool size is adjusted based on the usual throughput criteria.
+
+== When to tune the Open Liberty thread pool
+In most environments, configurations, and workloads, the Open Liberty thread pool does not require configuration or tuning.
+Open Liberty uses auto-tuning  to determine how many threads are needed to provide optimal server throughput.
+The thread pool controller continually adjusts the number of threads in the pool within the defined bounds for coreThreads and maxThreads.
+However, there are some situations in which setting the `coreThreads` or `maxThreads` attributes might be necessary.
+
+=== Threading settings
+
+The following configuration settings are available for the Open Liberty default executor pool:
+
+* `coreThreads` +
+This is the minimum number of threads in the pool (minimum 4).
+Open Liberty creates a new thread for each piece of offered work until the number of threads equals the value of this attribute.
+If `coreThreads` is not configured, it defaults to a multiple of the number of CPUs (hardware threads) available to the Open Liberty process.
+
+If Open Liberty is running in a shared environment, the thread pool controller cannot account for other processes with which it is sharing the available CPUs.
+In these cases, the default `coreThreads` value might cause Open Liberty to spin up more threads than is optimal, considering the other processes that are competing for CPU resources.
```

Comment:
this paragraph should be indented to line up under/as-part-of the coreThreads bullet item